src/engine/QueryPlanner.cpp

```diff
@@ -2275,6 +2287,12 @@ auto QueryPlanner::createSpatialJoin(const SubtreePlan& a, const SubtreePlan& b,
   }
 
   if (jcs.size() > 1) {
+    // If a spatial join operation substitutes a geometric relation filter,
+    // we might have multiple spatial joins for different pairs of variables
+    // that share some variable.
+    if (spatialJoin->getSubstitutesFilterOp()) {
+      return std::nullopt;
+    }
```

Comment:
The `throw` below can be mitigated in the future by simply applying a `FILTER`  in these cases. 
In particular, we might get the assertion triggered if we use the FILTER syntax.
But for now, maybe simply add a TODO there.